base/operators.jl

```diff
@@ -1287,17 +1287,25 @@ used to implement specialized methods.
 """
 in(x) = Fix2(in, x)
 
-function in(x, itr)
-    anymissing = false
-    for y in itr
-        v = (y == x)
-        if ismissing(v)
-            anymissing = true
-        elseif v
-            return true
+for ItrT = (Tuple,Any)
+    ex = quote
+        anymissing = false
+        for y in itr
+            v = (y == x)
+            if ismissing(v)
+                anymissing = true
+            elseif v
+                return true
+            end
         end
+        return anymissing ? missing : false
+    end
+    # define a generic method and a specialized version for `Tuple`,
+    # whose method bodies are identical, while adding better effects to the later
+    @eval function in(x, itr::$ItrT)
+        $(ItrT === Tuple ? :(@_terminates_locally_meta) : :nothing)
+        $ex
```

Comment:
I think it would be easier to read to paste the definition of `ex` here.

Reply:
Yeah, I agree. Expanded the method bodies there.